### PR TITLE
Implementation of Improved_UNET on ISIC data set for generating lesion segmentation

### PR DESCRIPTION
**Pull request by danielzhangau**
Name - Bosheng Zhang
Student ID - 45004830

This project is dedicated to achieving the Improved_UNET ISIC data set to generate predicted damage image segmentation, comparing the predicted and actual image segmentation using disease resistance, dice, and achieving a minimum dice score of 0.8 for the entire test data set. The train, validation, and test data sets are split at an 80:10:10 ratio. The dice loss and accuracy graphs of the training and validation data set wrt epoch are given.
The README.md file summarizes the overall goals and functions of each file.

### CODE REVIEW

**Comment by SiyuLiu0329** [comment]
- Some commit messages should be more specific
- Who are "we"? isn't this supposed to be individual work?
- Code valid for Normal Difficulty
- Good results

Other Feedback
- Please do not commit weights/checkpoints
- Please follow python file  and folder naming convention e.g. `lorem_ipsum_123_abc.py`

**Comment by danielzhangau** [comment]
Hi Siyu, 
Thanks for your feedback! I have updated the file name to follow the python file and folder naming convention. 
I am using 'we' is because I write it as a tutorial, people are like following me to learn how to use the files I write.

**Comment by rexsw** [comment]
Code looks good for normal difficulty

For next time;

Tensorflow has functions for image resizing and more so there’s no need to also use cv2 in your project. 

Comments such as 
    # input layer
     inputs = Input((img_size, img_size, 3))
Add nothing to the readability of the code and should be avoided. Overall comments were well done

Code needs more meaningful comments

Code in files like driver scripts  should be in a main function and in a python file.

**Comment by danielzhangau** [comment]
Thanks for your advice! I'll do what you suggested.

**Comment by rexsw** [comment]
You don't have too for this project (unless shakes request it) it's just some advice for the future =)

File: recognition/ISIC_skin_cancer/data_generator.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
+import os
```
Comment (shakes76) [inline recognition/ISIC_skin_cancer/data_generator.py:1]: Code header block should be added for all scripts/source files

**Comment by shakes76** [comment]
## TF Usage
OK
## Good Practice (Design/Commenting)
Good design and comments
## Algorithm 
Driver Script present
Shows Usage & Demo & Visualisation & Data usage, Readme OK
Module present
## Commit Log
Meaningful commit messages 
Progressive commits used 
## Documentation
Description and Comments good
Markdown used PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
**Feedback required. remove model files.**

File: recognition/ISIC_skin_cancer/data_generator.py (hunk 1 above)
Comment (danielzhangau) [reply recognition/ISIC_skin_cancer/data_generator.py:1]: Hi Shakes, 
I don't really get what do you mean about this comment. Could you please explain a bit more about it?

**Comment by danielzhangau** [comment]
Hi Shakes, 

Thanks for your feedback! I have removed the model file on the branch.

**Pull request merged**